src/extensions/default/CodeFolding/foldhelpers/handlebarsFold.js

```diff
@@ -0,0 +1,198 @@
+     */
+    function scanTextUntil(cm, startCh, startLine, condition) {
+        var line = cm.getLine(startLine),
+            seen = "",
+            characterIndex = startCh,
+            currentLine = startLine,
+            range;
+        while (currentLine <= cm.lastLine()) {
+            if (line.length === 0) {
+                characterIndex = 0;
+                line = cm.getLine(++currentLine);
+            } else {
+                seen = seen.concat(line[characterIndex] || "");
+                if (condition(seen)) {
+                    range = {
+                        from: {ch: startCh, line: startLine},
+                        to: {ch: characterIndex, line: currentLine},
+                        string: seen
+                    };
+                    return range;
+                } else if (characterIndex >= line.length) {
+                    seen = seen.concat(cm.lineSeparator());
+                    if (condition(seen)) {
+                        range = {
+                            from: {ch: startCh, line: startLine},
+                            to: {ch: characterIndex, line: currentLine},
+                            string: seen
+                        };
+                        return range;
+                    }
+                    characterIndex = 0;
+                    line = cm.getLine(++currentLine);
+                } else {
+                    ++characterIndex;
+                }
+            }
+        }
+    }
+
+    function endTag(seen) {
+        return endOfLineSpaceRegex.test(seen) || StringUtils.endsWith(seen, "}");
+    }
+
+    /**
+     * Returns a predicate function that returns true when a specific character is found
+     * @param   {string}   character the character to use in the match function
+     * @returns {function} A function that checks if the last character of the parameter string matches the parameter character
+     */
+    function readUntil(character) {
+        return function (seen) {
+            return seen[seen.length - 1] === character;
+        };
+    }
+
+    function getRange(cm, start) {
+        var currentLine = start.line,
+            text = cm.getLine(currentLine) || "",
+            i = 0,
+            tagStack = [],
+            braceStack = [],
+            found,
+            openTag,
+            openPos,
+            currentCharacter,
+            openTagIndex = text.indexOf("{{"),
+            range;
+
+        if (openTagIndex < 0 || text[openTagIndex + 2] === "/") {
+            return;
+        }
+
+        found = scanTextUntil(cm, openTagIndex + 2, currentLine, endTag);
+        if (!found) {
+            return;
+        }
+
+        openPos = {
+            from: {line: currentLine, ch: openTagIndex},
+            to: found.to
+        };
+        openTag = found.string.substring(0, found.string.length - 1);
+        if (openTag[0] === "#" || openTag[0] === "~") {
```

Comment:
Good point. I was working with the handlebars page as reference. `#*` should also be added.